### PR TITLE
fix xnnpack wasm build

### PR DESCRIPTION
**Pull request by guschmue**
fix xnnpack wasm build

### CODE REVIEW

File: cmake/external/xnnpack.cmake [hunk 1]
```diff
@@ -30,7 +30,13 @@ if (onnxruntime_BUILD_WEBASSEMBLY)
   endif()
 
   message("Adding WebAssembly Source Files to XNNPACK")
-  set(wasm_src_patterns "${XNNPACK_DIR}/src/wasm-*.c"
+  set(wasm_src_patterns "${XNNPACK_DIR}/src/init.c"
+                        "${XNNPACK_DIR}/src/params-init.c"
```
Comment (skottmckay) [inline cmake/external/xnnpack.cmake:34]: May be fighting a never ending battle here.

Should we instead have a script that parses the build.bazel file to extract the list of files?

Comment (skottmckay) [reply cmake/external/xnnpack.cmake:34]: This example should provide the bulk of the logic. I'm not a cmake expert so there may be a better way to do it.

```cmake
# parse the wasm source files from the bazel config
file(READ "${XNNPACK_DIR}/BUILD.bazel" xnnpack_bazel_config)

# Replace newlines with semicolon so the file contents can be converted to a list by CMake
# Also replace '[' and ']' so the bazel source lists don't get parsed as a nested list by cmake
string(REPLACE "\n" ";" xnnpack_bazel_config_lines "${xnnpack_bazel_config}")
string(REPLACE "[" "{" xnnpack_bazel_config_lines "${xnnpack_bazel_config_lines}")
string(REPLACE "]" "}" xnnpack_bazel_config_lines "${xnnpack_bazel_config_lines}")

set(_InWasmSection FALSE)
set(bazel_wasm_srcs "")

foreach(_line ${xnnpack_bazel_config_lines})
  if(NOT _InWasmSection)
    if(_line MATCHES "^ALL_WASM_MICROKERNEL_SRCS = \\{")
      set(_InWasmSection TRUE)
    endif()
  else()
    if(_line MATCHES "^\\}")
      set(_InWasmSection FALSE)
    else()
      # parse filename from quoted string with trailing comma
      string(REPLACE "\"" "" _line "${_line}")
      string(REPLACE "," "" _line "${_line}")
      string(STRIP "${_line}" _line)
      list(APPEND bazel_wasm_srcs "${XNNPACK_DIR}/${_line}")
    endif()
  endif()
endforeach()

message("Bazel Wasm Sources: ${bazel_wasm_srcs}")

```

There are 5 different wasm lists in the bazel so you may need to tweak to select the correct ones.

Options are:
  PROD_SCALAR_WASM_MICROKERNEL_SRCS
  ALL_WASM_MICROKERNEL_SRCS
  ALL_WASMSIMD_MICROKERNEL_SRCS
  ALL_WASMRELAXEDSIMD_MICROKERNEL_SRCS
  WASM32_ASM_MICROKERNEL_SRCS

Comment (guschmue) [reply cmake/external/xnnpack.cmake:34]: let me see if this works reliable

Comment (fs-eire) [reply cmake/external/xnnpack.cmake:34]: > May be fighting a never ending battle here.

I can't agree more on this idea. I definitely don't want to do this every time upgrading xnnpack version.

However, it is also important for now to fix the build ASAP to unblock other tasks. If it takes too long time to get the script working I would suggest to check in this PR first and do another PR to implement this idea.

**Comment by guschmue** [comment]
yeap, we'll merge this one and create another one based on the bazel file. Just waiting for the trt build to pass.

**Review by skottmckay — APPROVED** [review]

File: cmake/external/xnnpack.cmake (hunk 1 above)
Comment (guschmue) [reply cmake/external/xnnpack.cmake:34]: bazel/cmake is coming in a separate PR

**Pull request merged**